src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs

```diff
@@ -595,6 +595,7 @@ void IDisposable.Dispose()
 
                 responsesMcpTool.ServerDescription = mcpTool.ServerDescription;
                 responsesMcpTool.AuthorizationToken = mcpTool.AuthorizationToken;
+                responsesMcpTool.Headers = mcpTool.Headers;
```

Comment:
These headers should only be stored into Headers if mcpTool.Headers.Count > 0. Otherwise, it'll end up sending an empty collection in the JSON unnecessarily.